### PR TITLE
Remove tf_http_archive external Python deps

### PR DESCRIPTION
**Pull request by halseycamilla**
Migrate TF Python deps from being defined as tf_http_archives in workspace2.bzl to being in requirements.in. This consolidates where Python dependencies are defined, allows them to be more easily updated and also allows any version updates to be easily transferred to requirement.txt locked versions using the Hermetic Python requirements updater. Remove any downstream references to the deleted tf_http_archive repositories and if relevant replace them with @pypi_name//:pkg.

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @halseycamilla Can you please resolve conflicts? Thank you!

**Comment by halseycamilla** [comment]
> Hi @halseycamilla Can you please resolve conflicts? Thank you!

resolved!

**Comment by halseycamilla** [comment]
ROCm is failing but it seems as though it currently isn't in the TF .bazelrc file anymore so that should be expected. There is a separate PR rn in TF https://github.com/tensorflow/tensorflow/pull/61527 that appears to be trying to add it back.

File: tensorflow/opensource_only.files [hunk 1]
```diff
@@ -208,14 +206,17 @@ third_party/coremltools.BUILD:
...
 third_party/ducc/ducc0_custom_lowlevel_threading.h:
 third_party/ducc/threading.cc:
 third_party/ducc/threading.h:
 third_party/eigen3/BUILD:
 third_party/eigen3/Eigen/Cholesky:
 third_party/eigen3/Eigen/Core:
+third_party/ducc/BUILD:
+third_party/ducc/ducc0_custom_lowlevel_threading.h:
+third_party/ducc/threading.cc:
+third_party/ducc/threading.h:
```
Comment (MichaelHudgins) [inline tensorflow/opensource_only.files:219]: This appears to be adding a duplicate entry

**Review by MichaelHudgins — CHANGES_REQUESTED** [review]

File: tensorflow/opensource_only.files (hunk 1 above)
Comment (halseycamilla) [reply tensorflow/opensource_only.files]: Fixed

Comment (halseycamilla) [reply tensorflow/opensource_only.files]: Think it was a result of rebasing and also manually resolving so added both.

**Comment by gbaned** [comment]
Hi @halseycamilla Can you please resolve conflicts? Thank you!

**Pull request closed without merging**